data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/SourceCoordinator.java

```diff
@@ -49,9 +49,10 @@ public interface SourceCoordinator<T> {
      * @throws org.opensearch.dataprepper.model.source.coordinator.exceptions.PartitionNotOwnedException if the partition is not owned by this instance of SourceCoordinator
      * @throws org.opensearch.dataprepper.model.source.coordinator.exceptions.PartitionUpdateException if the partition can not be completed
      * @param partitionKey - The partition key that uniquely identifies the partition of work that was fully processed
+     * @param fromAcknowledgmentsCallback - Whether this method is being called from an acknowledgment callback or not
      * @since 2.2
      */
-    void completePartition(final String partitionKey);
+    void completePartition(final String partitionKey, final Boolean fromAcknowledgmentsCallback);
```

Comment:
Is there an object we can use here instead? Is there any acknowledgements handle of any sort? This would be ideal so that if data is needed from there in the future, we don't have to make yet another change to the interface.

Reply:
I don't think there's anything that makes sense to pass for acknowledgments other than the boolean. As far as objects in requests, it could make sense to keep the model class the same with

```
void completePartition(final CompletePartitionRequest)
``` 

and to do that with the rest of the SourceCoordinator methods as well. I'm not sure if that's worth a large refactor though

Reply:
I only suggested it if there is an existing model. I'm good keeping this as it is.